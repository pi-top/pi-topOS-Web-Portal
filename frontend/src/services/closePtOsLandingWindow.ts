import axios from "axios";

import apiBaseUrl from "./apiBaseUrl";

export default async function closePtOsLandingWindow() {
  await axios.post(
    `${apiBaseUrl}/close-pt-os-landing-window`,
    {},
  );
}
