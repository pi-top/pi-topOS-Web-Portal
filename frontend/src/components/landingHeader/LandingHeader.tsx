import React from 'react';
import styles from './LandingHeader.module.css';
import closeButtonImage from "../../assets/images/landing-close-button.svg";
import { runningOnWebRenderer } from '../../helpers/utils';
import MaskedDiv from '../atoms/masked/MaskedDiv';
import Button from '../atoms/button/Button';

import closePtOsLandingWindow from '../../services/closePtOsLandingWindow';


export default () => {
  return (
    <header className={styles.header}>
      <div className={styles.logoContainer}>
        <h1>
          Welcome to <span className="green">pi-topOS</span>
        </h1>
      </div>
      {runningOnWebRenderer() &&
        <MaskedDiv
          className={styles.closeButtonDiv}
          mask={`url(${closeButtonImage})`}
        >
          <Button
            className={styles.closeButton}
            onClick={() => closePtOsLandingWindow().catch(() => null)}
            aria-label="close-window"
          >
          </Button>
        </MaskedDiv>
      }
    </header>
  );
};
