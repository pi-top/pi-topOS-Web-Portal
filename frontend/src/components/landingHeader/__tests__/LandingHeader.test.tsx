import React from "react";
import { fireEvent, getByRole, render } from "@testing-library/react";

import LandingHeader from "../LandingHeader";
import closePtOsLandingWindow from "../../../services/closePtOsLandingWindow";

jest.mock("../../../services/closePtOsLandingWindow");

const closePtOsLandingWindowMock = closePtOsLandingWindow as jest.Mock;

let mockUserAgent = "any-browser";
Object.defineProperty(global.navigator, "userAgent", {
  get() {
    return mockUserAgent;
  },
});

describe("LandingHeader", () => {
  let layout: HTMLElement;
  let queryByAltText: any;
  let queryByText: any;
  let queryByLabelText: any;
  let getByText: any;
  let getByRole: any;
  let rerender: any;
  beforeEach(() => {
    closePtOsLandingWindowMock.mockResolvedValue("OK");
    ({
      container: layout,
      queryByAltText,
      queryByText,
      queryByLabelText,
      getByRole,
      getByText,
      rerender,
    } = render(<LandingHeader />));
  });

  it("renders header", () => {
    expect(layout.querySelector(".header")).toMatchSnapshot();
  });

  it("renders prompt message", () => {
    expect(layout.querySelector(".logoContainer")).toMatchSnapshot();
  });

  it("doesn't render close button (on regular browser)", () => {
    expect(layout.querySelector(".closeButtonDiv")).not.toBeInTheDocument();
  });

  describe("when running app on web-renderer", () => {
    beforeEach(() => {
        mockUserAgent = "web-renderer";
        rerender(<LandingHeader />);
    })

    it("renders close button (on regular browser)", () => {
      expect(layout.querySelector(".closeButtonDiv")).toBeInTheDocument();
    });

    it("closes landing app window when clicking close button", () => {
      fireEvent.click(getByRole("button", "close-window"));
      expect(closePtOsLandingWindowMock).toHaveBeenCalled();
    });

  });

});
