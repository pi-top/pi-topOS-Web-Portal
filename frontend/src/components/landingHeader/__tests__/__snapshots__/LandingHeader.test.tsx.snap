// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingHeader renders header 1`] = `
<header
  class="header"
>
  <div
    class="logoContainer"
  >
    <h1>
      Welcome to 
      <span
        class="green"
      >
        pi-topOS
      </span>
    </h1>
  </div>
</header>
`;

exports[`LandingHeader renders prompt message 1`] = `
<div
  class="logoContainer"
>
  <h1>
    Welcome to 
    <span
      class="green"
    >
      pi-topOS
    </span>
  </h1>
</div>
`;
